src/Plugin/MediaEntity/Type/WebdamAsset.php

```diff
@@ -156,7 +168,57 @@ public function getField(MediaInterface $media, $name) {
    * {@inheritdoc}
    */
   public function thumbnail(MediaInterface $media) {
-    // @TODO: Should this be a webdam thumbnail image?
+    if (isset($this->configuration['source_field'])) {
+      $source_field = $this->configuration['source_field'];
+      if ($media->hasField($source_field)) {
+        $property_name = $media->{$source_field}->first()->mainPropertyName();
+        $assetID = $media->{$source_field}->{$property_name};
+      }
+    }
+    // If we don't have an asset ID, there's not much we can do.
+    if (is_null($assetID)) {
+      return FALSE;
+    }
+    // Load the asset.
+    $asset = $this->webdam->getAsset($assetID);
+    //Download the webdam asset file as a string
+    $file_contents = $this->webdam->downloadAsset($asset->id);
+    //Set the path for webdam assets.
+    $path = 'public://webdam_assets/';
+    //Prepare webdam directory for writing and only proceed if successful
+    if(file_prepare_directory($path,FILE_CREATE_DIRECTORY)) {
+      //Save the file into Drupal
+      $file = file_save_data($file_contents, 'public://webdam_assets/' . $asset->id . '.' . $asset->filetype, FILE_EXISTS_REPLACE);
```

Comment:
Only thing here is to use whatever filesystem the file field is configured for. Hardcoding `public://` here (and a few lines up) isn't gonna work.

Reply:
Done.  Look for field setting if it exists.  Otherwise default to public